src/compile/data/optimizers.ts

```diff
@@ -78,3 +78,29 @@ export function removeUnusedSubtrees(node: DataFlowNode) {
   }
   return true;
 }
+
+/**
+ * Generates time unit nodes from selection filters and removes duplicate
+ * time unit nodes (as determined by the name of the output field)
+ * that may be generated due to selections projected over time units.
+ */
+export function cleanupSelectionTimeUnits(leaf: DataFlowNode) {
+  let fields = {};
+  return iterateFromLeaves((node: DataFlowNode) => {
+    if (node instanceof FilterNode) {
+      // Selection filters only add TimeUnitNodes on assembly.
```

Comment:
> With that said, the call to generating a selection predicate for a filter transform occurs in the expression method which is only called during FilterNode assembly. Can this be moved to parse time as well? I can try it soon if you're not sure off the top of your head.

You know more about that code than I do but from the dataflow perspective, I don't see any problems.

If you can move the expression when the selection gets parsed, then we have two options. Either we move selection parse before data parse so that we can insert the timeunit dataflow node when we parse the filter or we can add the dataflow node into the dataflow when we parse the selection. To do the latter, we would need to know, which selection filters rely on which selection, though. 

Overall, I would like to avoid inserting dataflow nodes during assemble.